### PR TITLE
typeck: show a note about tuple indexing for erroneous tup[i]

### PR DESCRIPTION
**Pull request by birkenfeld**
Fixes: #27842

### CODE REVIEW

File: src/test/compile-fail/issue-27842.rs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+fn main() {
+    let tup = (0, 1, 2);
+    // the case where we show a suggestion
+    let _ = tup[0];
+    //~^ ERROR cannot index a value of type
+    //~| HELP to access tuple elements, use tuple indexing syntax as shown
```
Comment (Manishearth) [inline src/test/compile-fail/issue-27842.rs]: compiletest supports SUGGESTION btw, could you add that entry here? r=me regardless

Comment (birkenfeld) [inline src/test/compile-fail/issue-27842.rs]: will do

**Pull request merged**